src/custom-lists/background/storage.ts

```diff
@@ -0,0 +1,212 @@
+    static CUSTOM_LISTS_COLL = 'customLists'
+    static LIST_ENTRIES_COLL = 'pageListEntries'
+
+    constructor({ storageManager }) {
+        super(storageManager)
+        this.storageManager.registerCollection(CustomListStorage.CUSTOM_LISTS_COLL, {
+            // different version for adding a new table.
+            version: new Date(2018, 6, 12),
+            fields: {
+                id: { type: 'string', pk: true },
+                name: { type: 'text' },
+                isDeletable: { type: 'binary' },
+                isNestable: { type: 'binary' },
+                createdAt: { type: 'datetime' },
+            },
+            indices: [
+                { field: 'id', pk: true, autoInc: true },
+                { field: 'name', unique: true },
+                { field: 'isDeletable' },
+                { field: 'isNestable' },
+                { field: 'createdAt' },
+            ],
+        })
+
+
+        this.storageManager.registerCollection(CustomListStorage.LIST_ENTRIES_COLL, {
+            // different version for adding a new table.
+            version: new Date(2018, 6, 12),
+            fields: {
+                listId: { type: 'string' },
+                pageUrl: { type: 'string' },
+                fullUrl: { type: 'string' },
+                createdAt: { type: 'datetime' },
+            },
+            indices: [
+                { field: ['listId', 'pageUrl'], pk: true },
+                { field: 'listId' },
+                { field: 'pageUrl' },
+            ],
+        })
+    }
+
+    // Takes the list as they come from Db and does some pre-processing before sending.
+    private changeListsBeforeSending(lists: PageList[], pageEntries: PageListEntry[]): PageList[] {
+        const mappedLists = lists.map((list) => {
+            const page = pageEntries.find(({ listId }) => listId === list.id)
+            delete list["_name_terms"]
+            return {
+                ...list,
+                // Set an empty pages array for manupulating in redux store.
+                pages: [],
+                active: Boolean(page),
+            }
+        })
+        return mappedLists
+    }
+
+    // Return all the list in the DB
+    //  TODO: Use pagination if required and decide correct types
+    async fetchAllLists({ query = {}, opts = {} }: { query?: any; opts?: any }) {
+        const x = await this.storageManager.findAll<PageList>(CustomListStorage.CUSTOM_LISTS_COLL, query, opts)
+        return this.changeListsBeforeSending(x, [])
+    }
+
+    async fetchListById(id: number) {
+        const list = await this.storageManager.findObject<PageList>(
+            CustomListStorage.CUSTOM_LISTS_COLL,
+            { id }
+        )
+        const pages = await this.storageManager.findAll<PageListEntry>(
+            CustomListStorage.LIST_ENTRIES_COLL,
+            {
+                listId: list.id
+            })
+        delete list["_name_terms"]
+        return list ? {
```

Comment:
I think this check needs to be before the second query. 
e.g.: If I give an `id` that doesn't exist, that first query should return nothing, but then when it attempts to perform the second query it will throw a `TypeError` as you're trying to access `list.id` on the result of the first query (L80).